### ISSUE TITLE
Model Parallelism

### ISSUE THREAD

**Issue by patrickvonplaten** [apple/ml-cross-entropy#33]
First of all, very nice library! Really cool to see the memory reduction when using CCE!

Are there any plans to support this loss also for model parallelism? 

More specifically, when sharding a LLM with a library like `fairscale`, one usually uses a MP-sharded CE loss, such as https://github.com/facebookresearch/fairscale/blob/bbce559da058ef34cd9da6d9d0d9320e7b380a07/fairscale/nn/model_parallel/cross_entropy.py#L29

**Comment by erikwijmans** [comment]
Hi @patrickvonplaten this was something I was thinking about also. Does the API in #38 look like it would fit the bill?

**Comment by patrickvonplaten** [comment]
API looks super nice - very intuitive and easy to plug into existing codebases! Just did a quick review, but nothing major.

Think most people that use one or both of MP, FSDP (and PP, CP) handle the relevant groups via PT's new `DeviceMesh` object: https://pytorch.org/tutorials/recipes/distributed_device_mesh.html so maybe having a quick code snippet that shows how to retrieve the correct MP group from this would be amazing.
